be/src/exec/olap_scanner.cpp

```diff
@@ -257,13 +257,19 @@ Status OlapScanner::_init_return_columns() {
 
 Status OlapScanner::get_batch(RuntimeState* state, RowBatch* batch, bool* eof) {
     // 2. Allocate Row's Tuple buf
+    const std::string mem_err_msg("OlapSanner::get_batch memory over limit, "
+                                  "please increase exec_mem_limit");
     uint8_t* tuple_buf =
-            batch->tuple_data_pool()->allocate(state->batch_size() * _tuple_desc->byte_size());
+            batch->tuple_data_pool()->try_allocate(state->batch_size() * _tuple_desc->byte_size());
+    if (tuple_buf == nullptr) {
+        return Status::MemoryLimitExceeded(mem_err_msg);
```

Comment:
<img width="939" alt="image" src="https://user-images.githubusercontent.com/13197424/158027472-17dd6f11-3451-4c05-9695-ce4708b650e5.png">
Can consider returning the status of mem limit exceeded in this way,
This will return more complete information including mem tracker statistics.